### PR TITLE
[CCUBE-1296][SR] Amended the code to ensure that the button is disabled when the input is invalid in prefill & validation.

### PR DESCRIPTION
**Pull request by ShIrInRuTh**
**Changes**

-  Amended the code to ensure that the button is disabled when the input is invalid in prefill & validation.

**Additional information**

-  Issue during deskcheck when showcasing was that the button was not disabled when the input was invalid.

### CODE REVIEW

File: src/components/element-editor/prefill/prefill.tsx [hunk 1]
```diff
@@ -41,8 +41,10 @@ export const Prefill = () => {
                 });
                 return !!validationResult ? false : true;
             } catch (error) {
-                return error.errors.some((errorMessage: string | string[]) =>
-                    errorMessage.includes("required")
+                return error.errors.some(
+                    (errorMessage: string | string[]) =>
+                        errorMessage.includes("required") ||
+                        errorMessage.includes("Invalid")
                 );
```
Comment (weili-govtech) [inline src/components/element-editor/prefill/prefill.tsx]: on second look, actually why do we need to check the content of the error message? shouldnt it be deemed erroneous as long as validation failed?

```suggestion
               return Yup.ValidationError.isError(error);
```

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/components/element-editor/prefill/prefill.tsx [hunk 2]
```diff
@@ -41,8 +41,10 @@ export const Prefill = () => {
                 });
                 return !!validationResult ? false : true;
```
Comment (weili-govtech) [inline src/components/element-editor/prefill/prefill.tsx]: can deem validation successful as long as `validateSync()` does not throw error

```suggestion
                return true;
```

File: src/components/element-editor/prefill/prefill.tsx (hunk 1 above)
Comment (ShIrInRuTh) [reply src/components/element-editor/prefill/prefill.tsx]: fixed in the last push

File: src/components/element-editor/prefill/prefill.tsx (hunk 2 above)
Comment (ShIrInRuTh) [reply src/components/element-editor/prefill/prefill.tsx]: fixed in the last push

File: src/components/element-editor/conditional-rendering/conditional-rendering.tsx [hunk 3]
```diff
@@ -65,11 +66,9 @@ export const ConditionalRendering = () => {
                     conditionalRendering: conditionalRenderingValues,
```
Comment (weili-govtech) [inline src/components/element-editor/conditional-rendering/conditional-rendering.tsx:66]: no need to declare `validationResult` as it is not used

```
validationSchema.validateSync({
    prefill: prefillValues,
    abortEarly: false,
});
```

**Review by weili-govtech — CHANGES_REQUESTED** [review]

**Review by weili-govtech — APPROVED** [review]

**Pull request merged**